plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -169,6 +169,23 @@ func findNodesThatFit(pod *api.Pod, nodeNameToInfo map[string]*schedulercache.No
 			if err != nil {
 				return api.NodeList{}, FailedPredicateMap{}, err
 			}
+			//add the extender failed info to failedPredicateMap
+			for _, filteredNode := range filtered {
```

Comment:
These two for loops seems like hackish workaround.
It's Extender.Filter() that should return the reason of why it failed some nodes.